### DISCUSSION TITLE
Question about the order of execution of uv_shutdown_cb and uv_write_cb

### DISCUSSION THREAD

**Discussion by zhaowq32** [libuv/libuv#4135] [Q&A] [answered]
When the callback function of uv_shutdown is called, whether the callback function of uv_write has been fully called?

**Comment by zhaowq32** [comment]
If I call uv_close, will the callback function of uv_write still be triggered?

**Comment by vtjnash** [comment, marked as answer]
uv_shutdown generally waits for pending writes to finish. The uv_close callback will always be the last callback for an object.

**Reply by zhaowq32** [reply]
Thanks for your reply.
